src/ceph-volume/ceph_volume/util/device.py

```diff
@@ -478,8 +484,15 @@ def _check_generic_reject_reasons(self):
             rejected.append("Device type is not acceptable. It should be raw device or partition")
         if self.is_ceph_disk_member:
             rejected.append("Used by ceph-disk")
-        if self.has_bluestore_label:
-            rejected.append('Has BlueStore device label')
+
+        try:
+            if self.has_bluestore_label:
+                rejected.append('Has BlueStore device label')
+        except OSError:
+            # likely failed to open the device. assuming it is BlueStore is the safest option
+            # so that a possibly-already-existing OSD doesn't get overwritten
+            rejected.append('Failed to determine if device is BlueStore')
```

Comment:
Should we also print the underlying error?

Reply:
I like the idea of logging the error. I'll do that.